initrd/etc/luks-functions

```diff
@@ -1,18 +1,129 @@
+		select_luks_container_size_percent
+		PERCENTAGE=$(cat /tmp/luks_container_size_percent)
+	fi
+
+
+	#Get disk size in bytes from fdisk
+	DISK_SIZE_BYTES=$(fdisk -l $DEVICE | grep Disk | awk -F " " {'print $5'}) > /dev/null 2>&1 || die "Error getting disk size from fdisk"
+	#Convert disk size to MB
+	DISK_SIZE_MB=$(calc $DISK_SIZE_BYTES/1024/1024 | awk -F "." {'print $1'}) > /dev/null 2>&1 || die "Error calculating disk size"
+	#Get size in bytes from percentage and apply percentage to DISK_SIZE_MB
+	PERCENTAGE_MB=$(calc $DISK_SIZE_MB*$PERCENTAGE/100 | awk -F "." {'print $1'}) > /dev/null 2>&1 || die "Error calculating percentage"
+	
+	#Console and whiptail $BG_COLOR_WARNING prompt (Y/n) validate one last time wiping and repartitioning of $device of total size $DISK_SIZE_MB with $PERCENTAGE_MB assigned to LUKS encrypted private partition
+	if [ -x /bin/whiptail ]; then
+		whiptail $BG_COLOR_WARNING --title "WARNING: Wiping and repartitioning $DEVICE of $DISK_SIZE_MB MB" --yesno \
+			"WARNING: Wiping and repartitioning $DEVICE with $PERCENTAGE_MB MB assigned to private LUKS contained private ext4 partition, rest assigned to extfat public partition.\n\nAre you sure you want to continue?" 0 80 \
+			|| die "User cancelled wiping and repartitioning of $DEVICE"
+	else
+		echo -e -n "Warning: Wiping and repartitioning $DEVICE with $PERCENTAGE_MB MB assigned to private LUKS contained private ext4 partition, rest assigned to extfat public partition.\n\nAre you sure you want to continue?"
+		read -r -p " [Y/n] " response
+		#transform response to uppercase with bash parameter expansion
+		response=${response^^}
+		#continue if response different then uppercase N
+		if [[ $response =~ ^(N)$ ]]; then
+			die "User cancelled wiping and repartitioning of $DEVICE"
+		fi
+	fi
+
+	echo -e "Preparing $DEVICE with $PERCENTAGE_MB MB for private LUKS container and rest of disk with exfat\
+	\n for public partition (This may take a while)..." | fold -s
+	DEBUG "Creating empty DOS partition table on device through fdisk to start clean"
+	echo -e "o\nw\n" | fdisk $DEVICE > /dev/null 2>&1 || die "Error creating partition table"
+	DEBUG "partition device with two partitions: first one being the percent applied and rest for second partition through fdisk"
+  	echo -e "n\np\n1\n\n+"$PERCENTAGE_MB"M\nn\np\n2\n\n\nw\n" | fdisk $DEVICE > /dev/null 2>&1 || die "Error partitioning device"
+	DEBUG "cryptsetup luksFormat  first partition with LUKS container aes-xts-plain64 cipher with sha256 hash and 512 bit key"
+	DEBUG "Creating ${PERCENTAGE_MB}MB LUKS container on ${DEVICE}1..."
+	echo -n "$PASSPHRASE" \
+		| cryptsetup -c aes-xts-plain64 -h sha256 -s 512 -y luksFormat ${DEVICE}1 > /dev/null 2>&1 \
+		|| die "Error formatting LUKS container"
+	DEBUG "Opening LUKS device and mapping under /dev/mapper/private..."
+	echo -n "$PASSPHRASE" | cryptsetup luksOpen ${DEVICE}1 private > /dev/null 2>&1 || die "Error opening LUKS container"
```

Comment:
Like above, open/close are preferred over luksOpen/luksClose, and `--keyfile -` is probably ideal

Reply:
@JonathonHall-Purism I discovered a new addition which permits to not have anything exposed:
`cryptsetup open "$USB_MOUNT_DEVICE" "usb_mount_$(basename "$USB_MOUNT_DEVICE")" --key-file <(echo -n "${PASS}")`